R/functions_io.R

```diff
@@ -464,3 +464,138 @@ ParsePlateInformation = function(cell_names, pattern='^(\\S+)_(\\d+)_([A-Z])(\\d
+  
+  crb_obj$samples[["by_cluster"]] = data.frame(sample = sc[[column_sample, drop=TRUE]], cluster=sc[[column_cluster, drop=TRUE]]) %>%
+    dplyr::group_by(sample,cluster) %>%
+    dplyr::summarise(num_cells=length(cluster)) %>%
+    dplyr::group_by(sample) %>%
+    dplyr::mutate(total_cell_count=sum(num_cells)) %>%
+    tidyr::pivot_wider(names_from=cluster, values_from=num_cells, values_fill=0)
+  
+  if ("Phase" %in% colnames(sc[[]])) {
+    crb_obj$samples[["by_cell_cycle_seurat"]] = data.frame(sample = sc[[column_sample, drop=TRUE]], phase=sc[[column_ccphase, drop=TRUE]]) %>% 
+      dplyr::group_by(sample,phase) %>%
+      dplyr::summarise(num_cells=length(phase)) %>%
+      dplyr::group_by(sample) %>%
+      dplyr::mutate(total_cell_count=sum(num_cells)) %>%
+      tidyr::pivot_wider(names_from=phase, values_from=num_cells, values_fill=0)
+  }
+  
+  # Add cluster-related information
+  crb_obj$clusters = list()
+  crb_obj$clusters[["colors"]] = param$col_clusters
+  crb_obj$clusters[["overview"]] = data.frame(cluster=levels(sc[[column_cluster, drop=TRUE]]))
+  
+  crb_obj$clusters[["by_samples"]] = data.frame(sample = sc[[column_sample, drop=TRUE]], cluster=sc[[column_cluster, drop=TRUE]]) %>%
+    dplyr::group_by(sample,cluster) %>%
+    dplyr::summarise(num_cells=length(cluster)) %>%
+    dplyr::group_by(cluster) %>%
+    dplyr::mutate(total_cell_count=sum(num_cells)) %>%
+    tidyr::pivot_wider(names_from=sample, values_from=num_cells, values_fill=0)
+  
+  if ("Phase" %in% colnames(sc[[]])) {
+    crb_obj$clusters[["by_cell_cycle_seurat"]] = data.frame(cluster = sc[[column_cluster, drop=TRUE]], phase=sc[[column_ccphase, drop=TRUE]]) %>% 
+      dplyr::group_by(cluster,phase) %>%
+      dplyr::summarise(num_cells=length(phase)) %>%
+      dplyr::group_by(cluster) %>%
+      dplyr::mutate(total_cell_count=sum(num_cells)) %>%
+      tidyr::pivot_wider(names_from=phase, values_from=num_cells, values_fill=0)
+  }
+  
+  tree = Tool(object=sc, slot="BuildClusterTree")
+  if(!is.null(tree)) crb_obj$clusters[["tree"]] = tree
+  
+  # Fix cell cycle information
+  if ("Phase" %in% colnames(sc[[]])) {
+    crb_obj$cells$cell_cycle_seurat = factor(crb_obj$cells$Phase,levels=c("G1","G2M","S"))
+  }
+  
+  # Add marker genes
+  crb_obj$marker_genes = list()
+  crb_obj$marker_genes[["parameters"]] = list(only_positive=FALSE, minimum_percentage=0.25, logFC_threshold=param$log2fc, test="MAST", p_value_threshold=param$padj)
+  if(!is.null(marker_genes) & nrow(marker_genes)>0) {
+    crb_obj$marker_genes[["by_cluster"]] = marker_genes %>% dplyr::select(c("cluster", "gene", "p_val", "avg_logFC", "pct.1", "pct.2", "p_val_adj"))
+  } else {
+    crb_obj$marker_genes[["by_cluster"]] = "no markers found"
+  }
+  crb_obj$marker_genes[["by_sample"]]
+  
+  # Add enriched pathways
+  crb_obj$enriched_pathways = list(enrichr=list())
+  crb_obj$enriched_pathways$enrichr$parameters = list(databases=param$enrichr_dbs, adj_p_cutoff="NA", max_terms="NA")
```

Comment:
Why "NA" here? Don't we have any cutoffs?

Reply:
Padj - not that I know. Max terms - I do not know. I think we return everything from enrichr

Reply:
Ok, I will change this in the future.